### DISCUSSION TITLE
FragmentHistogram Error:  in `combine_vars()`: ! Faceting variables must have at least one value.

### DISCUSSION THREAD

**Discussion by xiaoxiaoh16** [stuart-lab/signac#1520] [Q&A] [answered]
Hi @timoast, thanks fo the nice package, I have the result of Cellrager-atac output (filtered_peak_bc_matrix.h5, fragments.tsv.gz
fragments.tsv.gz.tbi, singlecell.csv) as the input of  Signac , I have some issues when I try to use FragmentHistogram function to plot. The error information and packages verison as following.

1. packages verison:
> packageVersion("Signac")
[1] '1.11.0'
> packageVersion("GenomeInfoDb")
[1] '1.36.4'
> packageVersion("Seurat")
[1] '4.4.0'
> packageVersion("GenomicFeatures")
[1] '1.52.2'
> packageVersion("ggplot2")
[1] '3.4.4'

2. Error message:
> FragmentHistogram(object = pbmc, group.by = 'nucleosome_group')
Error in `combine_vars()`:
! Faceting variables must have at least one value
Run `rlang::last_trace()` to see where the error occurred.
> 
> rlang::last_trace()
<error/rlang_error>
Error in `combine_vars()`:
! Faceting variables must have at least one value
---
Backtrace:
    ▆
 1. ├─base (local) `<fn>`(x)
 2. └─ggplot2:::print.ggplot(x)
 3.   ├─ggplot2::ggplot_build(x)
 4.   └─ggplot2:::ggplot_build.ggplot(x)
 5.     └─layout$setup(data, plot$data, plot$plot_env)
 6.       └─ggplot2 (local) setup(..., self = self)
 7.         └─self$facet$compute_layout(data, self$facet_params)
 8.           └─ggplot2 (local) compute_layout(..., self = self)
 9.             └─ggplot2::combine_vars(data, params$plot_env, vars, drop = params$drop)
Run rlang::last_trace(drop = FALSE) to see 2 hidden frames.
> 

3. Other information:
  > table(pbmc$nucleosome_group)

NS < 4 NS > 4 
  3811     67 
> 
> head(pbmc$nucleosome_signal)
AAACGAAAGTAATCAG-1 AAACGAACACGCTGTG-1 AAACGAATCCTGGGAC-1 AAACGAATCGGGAAAC-1 
         0.9176285          0.6792104          0.5595331          0.7097404 
AAACTCGAGAAAGCAG-1 AAACTCGAGAATAGTC-1 
         0.9242579         10.2473118 
> 

Please let me know if you need other information.
Thank you in advance!

**Comment by timoast** [comment, marked as answer]
Try increasing the size of the `region` parameter, probably there were no fragments found in the default region for the cells with `NS > 4`
